src/Elasticsearch.Net/Configuration/ConnectionConfiguration.cs

```diff
@@ -363,6 +363,25 @@ public T Proxy(Uri proxyAdress, string username, string password)
 		/// </summary>
 		public T EnableHttpPipelining(bool enabled = true) => Assign(a => a._enableHttpPipelining = enabled);
 
+		/// <summary>
+		/// Turns on settings that aid in debugging like DisableDirectStreaming() and PrettyJson()
+		/// and writes debug information to the output window.
+		/// </summary>
+		public T EnableDebugMode(Action<IApiCallDetails> onRequestCompleted = null)
+		{
+			this._disableDirectStreaming = true;
+			this._prettyJson = true;
+
+			var originalCompletedRequestHandler = this._completedRequestHandler;
+			this._completedRequestHandler = d =>
+			{
+				originalCompletedRequestHandler?.Invoke(d);
+				onRequestCompleted?.Invoke(d);
+				System.Diagnostics.Debug.WriteLine(d.DebugInformation);

```

Comment:
I sort of like that we write to it automatically as part of "debug mode", but I see the argument against it.  I'll remove it and update the PR 👍.